mergify_engine/rules/filter.py

```diff
@@ -491,3 +504,189 @@ def NearDatetimeFilter(
+                IncompleteChecksAny,
+                _identity,
+            ),
+            "<": (
+                cast_ret_to_incomplete_check_result(operator.lt),
+                IncompleteChecksAny,
+                _identity,
+            ),
+            ">": (
+                cast_ret_to_incomplete_check_result(operator.gt),
+                IncompleteChecksAny,
+                _identity,
+            ),
+            "<=": (
+                cast_ret_to_incomplete_check_result(operator.le),
+                IncompleteChecksAny,
+                _identity,
+            ),
+            ">=": (
+                cast_ret_to_incomplete_check_result(operator.ge),
+                IncompleteChecksAny,
+                _identity,
+            ),
+            "!=": (
+                cast_ret_to_incomplete_check_result(operator.ne),
+                IncompleteChecksAll,
+                _identity,
+            ),
+            "~=": (
+                cast_ret_to_incomplete_check_result(
+                    lambda a, b: a is not None and b.search(a)
+                ),
+                IncompleteChecksAny,
+                re.compile,
+            ),
+        }
+    )
+    multiple_operators: typing.Dict[
+        str, MultipleOperatorT[IncompleteChecksResult]
+    ] = dataclasses.field(
+        default_factory=lambda: {
+            "or": IncompleteChecksAny,
+            "and": IncompleteChecksAll,
+        }
+    )
+    pending_checks: typing.List[str] = dataclasses.field(default_factory=list)
+    all_checks: typing.List[str] = dataclasses.field(default_factory=list)
+
+    def _eval_binary_op(
+        self,
+        op: BinaryOperatorT[IncompleteChecksResult],
+        attribute_name: str,
+        attribute_values: typing.List[typing.Any],
+        ref_values_expanded: typing.List[typing.Any],
+    ) -> IncompleteChecksResult:
+        if not self.is_complete(op, attribute_name, ref_values_expanded):
+            return IncompleteCheck
+
+        binary_op, iterable_op, _ = op
+        return super()._eval_binary_op(
+            op, attribute_name, attribute_values, ref_values_expanded
+        )
+
+    def is_complete(
+        self,
+        op: BinaryOperatorT[FilterResultT],
+        attribute_name: str,
+        ref_values: typing.List[typing.Any],
+    ) -> bool:
+        binary_op, iterable_op, _ = op
+
+        if attribute_name.startswith(Filter.LENGTH_OPERATOR):
+            real_attr_name = attribute_name[1:]
+        else:
+            real_attr_name = attribute_name
+
+        if real_attr_name.startswith("check-") or real_attr_name.startswith("status-"):
+            if attribute_name.startswith(Filter.LENGTH_OPERATOR):
+                if len(self.pending_checks) != 0:
+                    return False
+            else:
+                if real_attr_name in (
+                    "check-pending",
+                    "check-success-or-neutral-or-pending",
```

Comment:
do we need `status-` variant too? L654 seems to handle it.

Reply:
status-pending doesn't exist.